crates/api_crud/src/post/create.rs

```diff
@@ -89,7 +89,10 @@ pub async fn create_post(
   // Fetch post links and pictrs cached image
   let metadata = fetch_link_metadata_opt(url.as_ref(), true, &context).await;
   let url = proxy_image_link_opt_apub(url, &context).await?;
-  let thumbnail_url = custom_thumbnail.map(Into::into).or(metadata.thumbnail);
+  let thumbnail_url = proxy_image_link_opt_apub(custom_thumbnail, &context)
+    .await?
+    .map(Into::into)
+    .or(metadata.thumbnail);
```

Comment:
Yep, the metadata thumbnail is already proxied.

Reply:
Even if a custom thumbnail is provided this will still generate a thumbnail from the url and store it in pictrs. Thats a waste of resources and should be skipped.

Reply:
Good call, fixed.